### ISSUE TITLE
Importing unneeded modules

### ISSUE THREAD

**Issue by torzdf** [deepfakes/faceswap#311]
Quite often I will find that I need to do a non-gpu related task whilst I am training, but I am unable to do so as so many modules get loaded at run time, regardless of whether they are required for the specific task resulting in an oom error.

I know with python it is general practice to import modules at the beginning of the script, but with those that load GPU related modules, would it not make more sense to load them when required? For example, the face filter loads face_recognition as part of the argparse import regardless of whether it is going to be used or not, leading to a potential cudaMalloc oom error if training is running as it attempts to instigate dlib. This happens even if you were to enter:
`python faceswap.py -h`

I'm happy to go through the code and make this change, but wanted to check whether this would be frowned upon, or whether there is a good reason I should not do this before embarking on the task.

**Comment by Clorr** [comment]
Yes that would be a nice improvment

**Comment by bryanlyon** [comment]
I agree that could be helpful, but traversing a while file for all the imports is a tad annoying.   Can you keep them at the top through a function or something so that they're easy to find.

**Comment by torzdf** [comment]
I will. There's only a couple which need moving really. Already made a start. Not fussed about all of them, just those that utilize the GPU.
